### PR TITLE
docs: update docs for managed identity

### PR DESCRIPTION
**Pull request by aramase**
Signed-off-by: Anish Ramasekar <anish.ramasekar@gmail.com>

**Reason for Change**:

- Updates docs to reflect managed identity support with workload identity
  - Modifies quick start to include steps for managed identity
  - Removes managed identity limitation from FAQ
  - Adds steps for creating/deleting FIC for managed identity
  - Portal experience for setting up FIC with MI

**Requirements**

- [ ] squashed commits
- [ ] included documentation
- [ ] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

part of #325 

**Please answer the following questions with yes/no**:

Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

- [ ] yes
- [ ] no

**Notes for Reviewers**:
Pending items (to be added in follow-up PRs):
- Migration guide using proxy sidecar

### CODE REVIEW

File: docs/book/src/topics/federated-identity-credential.md [hunk 1]
```diff
@@ -84,14 +111,32 @@ az ad app federated-credential delete --federated-credential-id $FIC_ID --id $AP
...
 1. Specify the **Namespace**.
 1. Specify the **Service account name**.
-1. The **Subject identifier** field autopopulate based on the values you entered.
+1. The **Subject identifier** field autopopulates based on the values you entered.
 1. Add a **Name** for the federated credential.
 1. Click **Add** to configure the federated credential.
 
 ![Screenshot showing Azure Portal app registration federated credential screen for Kubernetes scenario](../images/azure-portal-federated-credential-kubernetes.png)
 
+### Federated identity credential for an User-assigned managed identity
```
Comment (sozercan) [inline docs/book/src/topics/federated-identity-credential.md:120]:
```suggestion
### Federated identity credential for a user-assigned managed identity
```

File: docs/book/src/topics/federated-identity-credential.md [hunk 2]
```diff
@@ -74,8 +85,24 @@ Select the desired ID of the federated identity credential and run the following
 az ad app federated-credential delete --federated-credential-id $FIC_ID --id $APPLICATION_OBJECT_ID
 ```
 
+### Federated identity credential for an User-assigned managed identity
+
+```bash
+az identity federated-credential create --name "kubernetes-federated-identity" --identity-name "${USER_ASSIGNED_MANAGED_IDENTITY_NAME}" --resource-group "${RESOURCE_GROUP}" --issuer "${SERVICE_ACCOUNT_ISSUER}" --subject "system:serviceaccount:${SERVICE_ACCOUNT_NAMESPACE}:${SERVICE_ACCOUNT_NAME}"
```
Comment (sozercan) [inline docs/book/src/topics/federated-identity-credential.md:91]:
```suggestion
az identity federated-credential create \
	--name "kubernetes-federated-identity" \
	--identity-name "${USER_ASSIGNED_MANAGED_IDENTITY_NAME}" \
	--resource-group "${RESOURCE_GROUP}" \
	--issuer "${SERVICE_ACCOUNT_ISSUER}" \
	--subject "system:serviceaccount:${SERVICE_ACCOUNT_NAMESPACE}:${SERVICE_ACCOUNT_NAME}"
```

File: docs/book/src/topics/federated-identity-credential.md [hunk 3]
```diff
@@ -74,8 +85,24 @@ Select the desired ID of the federated identity credential and run the following
...
+```
+
+To delete a federated identity credential 
+
+```bash
+az identity federated-credential delete --identity-name "${USER_ASSIGNED_MANAGED_IDENTITY_NAME}" --name "kubernetes-federated-identity" --resource-group "${RESOURCE_GROUP}"
```
Comment (sozercan) [inline docs/book/src/topics/federated-identity-credential.md:97]:
```suggestion
az identity federated-credential delete \
	--identity-name "${USER_ASSIGNED_MANAGED_IDENTITY_NAME}" \
	--name "kubernetes-federated-identity" \
	--resource-group "${RESOURCE_GROUP}"
```

File: docs/book/src/quick-start.md [hunk 4]
```diff
@@ -9,7 +9,7 @@ Before we get started, ensure the following:
 * Azure CLI version 2.39.0 or higher. Run `az --version` to verify.
 *  You are logged in with the Azure CLI as a user.
    *  If you are logged in with a Service Principal, ensure that it has the correct [API permissions][14] enabled.
-*  Your logged in account must have sufficient permissions to create applications and service principals in Azure AD.
+*  Your logged in account must have sufficient permissions to create applications and service principals or user-assigned managed identity in Azure AD.
```
Comment (sozercan) [inline docs/book/src/quick-start.md:12]:
```suggestion
*  Your logged in account must have sufficient permissions to create applications and service principals or user-assigned managed identities in Azure AD.
```

File: docs/book/src/installation/mutating-admission-webhook.md [hunk 5]
```diff
@@ -5,12 +5,12 @@ Azure AD Workload Identity uses a [mutating admission webhook][1] to project a s
...
-| ---------------------------- | ------------------------------------------------------ |
-| `AZURE_AUTHORITY_HOST`       | The Azure Active Directory (AAD) endpoint.             |
-| `AZURE_CLIENT_ID`            | The application/client ID of the Azure AD application. |
-| `AZURE_TENANT_ID`            | The tenant ID of the Azure account.                    |
-| `AZURE_FEDERATED_TOKEN_FILE` | The path of the projected service account token file.  |
+| Environment variable         | Description                                                                              |
+| ---------------------------- | ---------------------------------------------------------------------------------------- |
+| `AZURE_AUTHORITY_HOST`       | The Azure Active Directory (AAD) endpoint.                                               |
+| `AZURE_CLIENT_ID`            | The application/client ID of the Azure AD application or user-assigned managed identity. |
+| `AZURE_TENANT_ID`            | The tenant ID of the Azure account.                                                      |
```
Comment (sozercan) [inline docs/book/src/installation/mutating-admission-webhook.md:12]:
```suggestion
| `AZURE_TENANT_ID`            | The tenant ID of the Azure subscription.                                                      |
```

File: docs/book/src/quick-start.md [hunk 6]
```diff
@@ -9,7 +9,7 @@ Before we get started, ensure the following:
 * Azure CLI version 2.39.0 or higher. Run `az --version` to verify.
```
Comment (sozercan) [inline docs/book/src/quick-start.md:9]: should we use 2.40.0?

File: docs/book/src/topics/federated-identity-credential.md [hunk 7]
```diff
@@ -84,14 +111,32 @@ az ad app federated-credential delete --federated-credential-id $FIC_ID --id $AP
 1. Specify the **Cluster issuer URL**.
 1. Specify the **Namespace**.
 1. Specify the **Service account name**.
-1. The **Subject identifier** field autopopulate based on the values you entered.
+1. The **Subject identifier** field autopopulates based on the values you entered.
```
Comment (sozercan) [inline docs/book/src/topics/federated-identity-credential.md:114]:
```suggestion
1. The **Subject identifier** field auto-populates based on the values you entered.
```

File: docs/book/src/topics/federated-identity-credential.md [hunk 8]
```diff
@@ -84,14 +111,32 @@ az ad app federated-credential delete --federated-credential-id $FIC_ID --id $AP
...
+
+1. Sign in to the [Azure portal](https://portal.azure.com). 
+1. Go to resource group and open the user-assigned managed identity you want to configure.
+1. Go to **Federated credentials**.
+1. In the **Federated credentials** tab, select **Add credential**. The **Add a credential** blade opens.
+1. In the **Federated credential scenario** drop-down box select **Kubernetes accessing Azure resources**.
+1. Specify the **Cluster issuer URL**.
+1. Specify the **Namespace**.
+1. Specify the **Service account name**.
+1. The **Subject identifier** field autopopulates based on the values you entered.
```
Comment (sozercan) [inline docs/book/src/topics/federated-identity-credential.md:130]:
```suggestion
1. The **Subject identifier** field auto-populates based on the values you entered.
```

File: docs/book/src/quick-start.md (hunk 6 above)
Comment (aramase) [reply docs/book/src/quick-start.md:9]: good catch!

**Review by sozercan — APPROVED** [review]

**Pull request merged**